subsys/bluetooth/audio/tbs_client.c

```diff
@@ -600,6 +599,85 @@ static uint8_t notify_handler(struct bt_conn *conn,
 	return BT_GATT_ITER_CONTINUE;
 }
 
+/* Common function to read tbs_client strings which may require long reads */
+static uint8_t handle_string_long_read(struct bt_conn *conn, uint8_t err,
+				       struct bt_gatt_read_params *params,
+				       const void *data,
+				       uint16_t length,
+				       bt_tbs_client_read_string_cb cb,
+				       bool truncatable)
+{
+	struct bt_tbs_instance *inst = CONTAINER_OF(params,
+						    struct bt_tbs_instance,
+						    read_params);
+	uint16_t offset = params->single.offset;
+	uint8_t inst_index = tbs_index(conn, inst);
+	const char *received_string;
+	int tbs_err = err;
+
+	if ((tbs_err == 0) && (data != NULL) &&
+	    (net_buf_simple_tailroom(&inst->net_buf) < length)) {
+		LOG_DBG("Read length %u: String buffer full", length);
+		if (truncatable) {
+			/* Use the remaining buffer and continue reading */
+			LOG_DBG("Truncating string");
+			length = net_buf_simple_tailroom(&inst->net_buf);
+		} else {
+			tbs_err = BT_ATT_ERR_INSUFFICIENT_RESOURCES;
+		}
+	}
+
+	if (tbs_err != 0) {
```

Comment:
Did you consider putting this check first? Since it covers all cases where `tbs_err != 0` you could then also remove `tbs_err = 0` from the other check.
Its maybe a nitpic, but I feel it would increase readability too, as both the check above and below `tbs_err != 0` actually handles the string.

Reply:
Disregard this. The top block can produce an error, which is handled here :)